models.py

```diff
@@ -108,98 +66,155 @@ def transform(self, embedding_matrix, X, y=None, validation_data=None):
         return {'prediction_probability': predictions}
 
 
-class GloveLSTM(GloveBasic):
-    def _build_optimizer(self, **kwargs):
-        return Adam(kwargs['lr'])
-
-    def _build_model(self, embedding_matrix, embedding_size,
-                     maxlen, max_features,
-                     unit_nr, repeat_block, dropout_lstm,
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_dense, use_prelu, use_batch_norm, trainable_embedding, global_pooling):
-        return lstm(embedding_matrix, embedding_size,
-                    maxlen, max_features,
-                    unit_nr, repeat_block, dropout_lstm,
-                    dense_size, repeat_dense, dropout_dense,
-                    l2_reg_dense, use_prelu, use_batch_norm, trainable_embedding, global_pooling)
-
-
-class GloveSCNN(GloveBasic):
-    def _build_optimizer(self, **kwargs):
-        return SGD(**kwargs)
-
-    def _build_model(self, embedding_matrix, embedding_size,
-                     maxlen, max_features,
-                     filter_nr, kernel_size, dropout_convo,
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_convo, l2_reg_dense, use_prelu, trainable_embedding, use_batch_norm):
-        return scnn(embedding_matrix, embedding_size,
-                    maxlen, max_features,
-                    filter_nr, kernel_size, dropout_convo,
-                    dense_size, repeat_dense, dropout_dense,
-                    l2_reg_convo, l2_reg_dense, use_prelu, trainable_embedding, use_batch_norm)
-
-
-class GloveDPCNN(GloveBasic):
-    def _build_optimizer(self, **kwargs):
-        return SGD(**kwargs)
-
-    def _build_model(self, embedding_matrix, embedding_size,
-                     maxlen, max_features,
-                     filter_nr, kernel_size, repeat_block, dropout_convo,
-                     dense_size, repeat_dense, dropout_dense,
-                     l2_reg_convo, l2_reg_dense, use_prelu, trainable_embedding, use_batch_norm):
+class WordSCNN(PretrainedEmbeddingModel):
+    def _build_model(self, embedding_matrix, embedding_size, trainable_embedding, maxlen, max_features,
```

Comment:
@jakubczakon I like the idea of using the `Embedding` layer in models: WordSCNN, WordDPCNN, WordCuDNNLSTM and WordCuDNNGRU. Since in Keras it must be first layer it is worth to consider making it part of steps as well. Your class `PretrainedEmbeddingModel` that modifies `fit` and `transform`and adds `embedding_matrix` may be moved to the steps -> keras -> models.py.

I think that models may also be added to steps, because they are not specifically designed to the toxic challenge.